src/BuildScriptGenerator/DefaultBuildScriptGenerator.cs

```diff
@@ -276,6 +276,35 @@ private void LogScriptIfGiven(string type, string scriptPath)
             var copySourceDirectoryContentToDestinationDirectory = buildScriptSnippets.All(
                 snippet => snippet.CopySourceDirectoryContentToDestinationDirectory);
 
+            string oryxAppType = string.Empty;
+            if (buildProperties != null
+                && buildProperties.TryGetValue(Constants.OryxAppType, out oryxAppType))
+            {
+                if (!string.IsNullOrEmpty(oryxAppType)
+                    || !string.IsNullOrWhiteSpace(oryxAppType))
+                {
+                    _logger.LogDebug($"Build Property Key {Constants.OryxAppType} with value {oryxAppType} is written into manifest");
+                    buildProperties[Constants.OryxAppType] = oryxAppType;
+                }
+            }
+            else if (context.Properties != null
+                && context.Properties.TryGetValue(Constants.OryxAppType, out oryxAppType))
+            {
+                if (!string.IsNullOrEmpty(oryxAppType)
+                    || !string.IsNullOrWhiteSpace(oryxAppType))
+                {
+                    _logger.LogDebug($"Context Property Key {Constants.OryxAppType} with value {oryxAppType} is written into manifest");
+                    buildProperties[Constants.OryxAppType] = oryxAppType;
+                }
+            }
+            else if (!string.IsNullOrEmpty(Environment.GetEnvironmentVariable(Constants.OryxAppType))
```

Comment:
If you recall we already have a system to deal with hierarchical config and that is through .NET Core's Configuration API, so we want to stick to it and also the above code does not handle content coming from `build.env`. You can take a look at line 410 in `BuildCommand.cs` on how we do this for `RequiredOsPackages`...so basically the value set from `-p` overrides value set via environment variable which in turn overrides value set in `build.env`.

Reply:
`build.env` is our own file...so i am not sure if you will find an example of it...sure, let us discuss about it